src/codegen/compile.ml

```diff
@@ -10579,7 +11012,7 @@ and compile_prim_invocation (env : E.t) ae p es at =
     SR.Vanilla,
     compile_exp_vanilla env ae e ^^
     Arr.len env ^^
-    BigNum.from_word30 env
+    BigNum.from_word32 env
```

Comment:
Good catch: If we would have relaxed the 2 billion element (2 * 1024 * 1024 * 1024) array length limit in `alloc_array` in the RTS, this would have been broken...

Reply:
Yep.